Automattic/wp-calypso
# Unify Design Preview: Support mobile layout with the navigator

Related to https://github.com/Automattic/wp-calypso/issues/78419

## Proposed Changes

* `@automattic/design-preview`: Now it can deal with the height by itself.
  * On the main screen (with navigator buttons), the height grows automatically
  * On the other screens, the height is the same as the viewport height reducing the height of the navigator footer. Also, the navigator screen will stick to the top of the screen on mobile.
* `DeviceSwitcher`: Introduce a new prop called `isFullscreen` to eliminate the max width/height
* Adjust lots of the styles to support mobile layout.

**Screenshots**

| Main | Bottom of Main | Colors | Fonts |
| - | - | - | - |
| ![image](https://github.com/Automattic/wp-calypso/assets/13596067/a7bb4ce7-c210-4334-9b01-8b0be45f4786) | ![image](https://github.com/Automattic/wp-calypso/assets/13596067/97b6cbd5-38d4-4823-90b4-b1275f7cfab9) | ![image](https://github.com/Automattic/wp-calypso/assets/13596067/835777ca-922e-4b47-992d-4c5210694d56) | ![image](https://github.com/Automattic/wp-calypso/assets/13596067/c0382f21-3ff1-4609-a52b-9306cba89a61) |

## Testing Instructions



* Go to the URLs below, especially for the mobile screen
  * /setup?siteSlug=<your_site>: Fonts only
  * /setup?siteSlug=<your_site>&flags=signup/design-picker-preview-colors: Colors & Fonts
* Continue until you land on the Design Picker
* Preview the design with style variation, everything should be good as before
* Preview the design without the style variation, and ensure you're able to navigate between screens

## Pre-merge Checklist
